services/worker-manager/src/providers/aws.js

```diff
@@ -82,83 +64,108 @@ class AwsProvider extends Provider {
       });
     }
 
-    // Make sure we don't get "The same resource type may not be specified more than once in tag specifications" errors
-    const TagSpecifications = config.launchConfig.TagSpecifications ? config.launchConfig.TagSpecifications : [];
-    const instanceTags = [];
-    const otherTagSpecs = [];
-    TagSpecifications.forEach(ts =>
-      ts.ResourceType === 'instance' ? instanceTags.concat(ts.Tags) : otherTagSpecs.push(ts)
-    );
+    const uniqueRegions = new Set(workerPool.config.launchConfigs.map(lc => lc.region));
+    let ec2s = {};
+    uniqueRegions.forEach(region => {
+      ec2s[region] = new aws.EC2({credentials: this.providerConfig.credentials, region});
+    });
 
-    let spawned;
-    try {
-      spawned = await ec2.runInstances({
-        ...config.launchConfig,
-
-        UserData: userData.toString('base64'), // The string needs to be base64-encoded. See the docs above
-
-        MaxCount: toSpawn,
-        MinCount: toSpawn,
-        TagSpecifications: [
-          ...otherTagSpecs,
-          {
-            ResourceType: 'instance',
-            Tags: [
-              ...instanceTags,
-              {
-                Key: 'CreatedBy',
-                Value: `taskcluster-wm-${this.providerId}`,
-              }, {
-                Key: 'Owner',
-                Value: workerPool.owner,
-              },
-              {
-                Key: 'ManagedBy',
-                Value: 'taskcluster',
-              },
-              {
-                Key: 'Name',
-                Value: `${workerPoolId}`,
-              }],
-          },
-        ],
-      }).promise();
-    } catch (e) {
-      this.monitor.err(`Error calling AWS API: ${e}`);
+    for await (let config of workerPool.config.launchConfigs) {
 
-      return await workerPool.reportError({
-        kind: 'creation-error',
-        title: 'Instance Creation Error',
-        description: e.message,
-        notify: this.notify,
-        WorkerPoolError: this.WorkerPoolError,
+      const toSpawn = await this.estimator.simple({
+        workerPoolId,
+        minCapacity: config.minCapacity,
+        maxCapacity: config.maxCapacity,
+        capacityPerInstance: config.capacityPerInstance,
+        running: workerPool.providerData[this.providerId].running, // todo: if this is per config now, do we need a config id?
```

Comment:
We'll need to add a `pending` to this as well and we can update pending from within this loop. I'm doing that now in my branch.

Reply:
hm, pending is considered `running`, isn't it?